### ISSUE TITLE
The HOME or COMPOSER_HOME environment variable 

### ISSUE THREAD

**Issue by Skelos** [composer/composer#6192]
My `composer.json`:

```json
I don't have one...
```

Output of `composer diagnose`:

```
sh: composer: not found
```

When I run this command:

```
php composer-setup.php
```

I get the following output:

```
All settings correct for using Composer
The HOME or COMPOSER_HOME environment variable must be set for composer to run correctly
```

And I expected this to happen:

'Installation successfully'

My architecture is:
Hardware : Cubox-i 4x4 (Quadcore ARM v7)
Software : OS = Geeexbox (linux) , webserver = lighttpd , php = PHP Version 5.4.12

**Comment by Skelos** [comment]
Could I ask you because you closed it? I looked here and on web for same problem with same architecture but I nothing found...

**Comment by alcohol** [comment]
The error describes the problem very clearly.

**Comment by curry684** [comment]
Geexbox is, [by its own description](http://www.geexbox.org/software-components-of-geexbox/), _"a free Linux distribution providing a media center software suite for personal computers and empbedded device. It is no based on other Linux distros but it’s complete build system from sources."_

It's likely by no means a feature-complete Linux distro fit to run development tools, and for performance apparently skims on a lot of standard Linux features like setting a `HOME` env variable. That's its choice, but Composer does need a place to store its data. Therefore, if the system does not provide a `HOME` variable, you can set the `COMPOSER_HOME` variable at will.

As alcohol already mentions the error you get is extremely clear and verbose. If you cannot fix it from there in all honesty you shouldn't be trying to install Composer on a specific-purpose micro-distro and just get Ubuntu or Mint. It'll work fine out of the box on both.

**Comment by Skelos** [comment]
Thank you alcohol and curry684. Very much.
I understood what is my fault and I wrote here hoping it could be useful for others.

The error I posted is generated trying to install composer via  [webconsole](http://web-console.org/).
So, I try to install composer on Geexbox via telnet and it was installed succesfully.

Conclusion: 
1) Using webconsole for install composer is a bad choice
2) Composer can be succesfully installed in Geexbox.
